lib/EarlyFunctionAnalysis/FunctionSummaryOracle.cpp

```diff
@@ -150,11 +150,14 @@ void importModel(llvm::Module &M,
   revng_log(Log, "Importing from model");
   LoggerIndent Indent(Log);
 
-  std::set<llvm::GlobalVariable *> ABICSVs;
-  for (llvm::GlobalVariable *CSV : GCBI.abiRegisters())
-    if (CSV != nullptr && !(GCBI.isSPReg(CSV)))
-      ABICSVs.emplace(CSV);
-  PrototypeImporter Importer{ .M = M, .ABICSVs = ABICSVs };
+  auto IsAnABICSV = [SP = GCBI.spReg()](llvm::GlobalVariable *CSV) -> bool {
```

Comment:
As before, I'd prefer something like `| keepABICSVsOnly` or something like this.

Reply:
Done, except I went with `RegisterFilter`.
```
  PrototypeImporter Importer{ .M = M,
                              .ABICSVs = GCBI.abiRegisters() | RegisterFilter
                                         | revng::to<std::set<GV *>>() };
```